Python/specialize.c

```diff
@@ -637,3 +640,45 @@ _Py_Specialize_LoadGlobal(
     cache0->counter = saturating_start();
     return 0;
 }
+
+int
+_Py_Specialize_BinarySubscr(
+     PyObject *container, PyObject *sub,
+    _Py_CODEUNIT *instr, SpecializedCacheEntry *cache)
+{
+    _PyAdaptiveEntry *cache0 = &cache->adaptive;
+    if (PyList_CheckExact(container)) {
```

Comment:
Does someone know if compilers are able to build a jump table if we use a switch on the object type? Example:

```
switch (Py_TYPE(container)) {
case &PyList_Type: ...
...
default: ...
}
```

Reply:
At least, you can get the type only once and then replace "check exact" tests with comparison with the types:

```
PyTypeObject *type = Py_TYPE(container);
if (type == &PyList_Type) ...
...
```

Reply:
> Does someone know if compilers are able to build a jump table if we use a switch on the object type?

Probably not. And if even if they did, it would almost certainly stop working once we have one `list` object per interpreter.